httpie/core.py

```diff
@@ -143,7 +143,7 @@ def program(args, env, log_error):
                 else:
                     raise
 
-        if downloader and exit_status == ExitStatus.OK:
+        if downloader:
```

Comment:
Please keep pull requests to a single change only. This seems to be related to another issue, so you can create a new PR for that.